source/common.c

```diff
@@ -352,6 +353,210 @@ void aws_secure_zero(void *pBuf, size_t bufsize) {
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_INVALID_ARGUMENT,
+        "An invalid argument was passed to a function."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_RANDOM_GEN_FAILED,
+        "A call to the random number generator failed. Retry later."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_MALFORMED_INPUT_STRING,
+        "An input string was passed to a parser and the string was incorrectly formatted."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_UNIMPLEMENTED,
+        "A function was called, but is not implemented."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_INVALID_STATE,
+        "An invalid state was encountered."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_ENVIRONMENT_GET,
+        "System call failure when getting an environment variable."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_ENVIRONMENT_SET,
+        "System call failure when setting an environment variable."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_ENVIRONMENT_UNSET,
+        "System call failure when unsetting an environment variable."
+    ),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_SYS_CALL_FAILURE,
+        "System call failure"),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_FILE_INVALID_PATH,
+        "Invalid file path."),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_MAX_FDS_EXCEEDED,
+        "The maximum number of fds has been exceeded."),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_NO_PERMISSION,
+        "User does not have permission to perform the requested action."),
+    AWS_DEFINE_ERROR_INFO_COMMON(
+        AWS_ERROR_STREAM_UNSEEKABLE,
+        "Stream does not support seek operations"),
+};
+/* clang-format on */
+
+static struct aws_error_info_list s_list = {
+    .error_list = errors,
+    .count = AWS_ARRAY_SIZE(errors),
+};
+
+static struct aws_log_subject_info s_common_log_subject_infos[] = {
+    DEFINE_LOG_SUBJECT_INFO(
+        AWS_LS_COMMON_GENERAL,
+        "aws-c-common",
+        "Subject for aws-c-common logging that doesn't belong to any particular category"),
+    DEFINE_LOG_SUBJECT_INFO(
+        AWS_LS_COMMON_TASK_SCHEDULER,
+        "task-scheduler",
+        "Subject for task scheduler or task specific logging."),
+};
+
+static struct aws_log_subject_info_list s_common_log_subject_list = {
+    .subject_list = s_common_log_subject_infos,
+    .count = AWS_ARRAY_SIZE(s_common_log_subject_infos),
+};
+
+static bool s_common_library_initialized = false;
+
+void aws_common_library_init(struct aws_allocator *allocator) {
+    (void)allocator;
+
+    if (!s_common_library_initialized) {
+        s_common_library_initialized = true;
+        aws_register_error_info(&s_list);
+        aws_register_log_subject_info_list(&s_common_log_subject_list);
+    }
+}
+
+void aws_common_library_clean_up(void) {
+    if (s_common_library_initialized) {
```

Comment:
Should it be an error (or at least a warning) to call clean_up on a library that was never init?

Reply:
No, the goal is for these functions to be completely idempotent.